examples/with-evm-web3-react-next/chains.ts

```diff
@@ -0,0 +1,164 @@
+  chainInformation: BasicChainInformation | ExtendedChainInformation
+): chainInformation is ExtendedChainInformation {
+  return !!(chainInformation as ExtendedChainInformation).nativeCurrency;
+}
+
+export function getAddChainParameters(
+  chainId: number
+): AddEthereumChainParameter | number {
+  const chainInformation = CHAINS[chainId];
+  if (isExtendedChainInformation(chainInformation)) {
+    return {
+      chainId,
+      chainName: chainInformation.name,
+      nativeCurrency: chainInformation.nativeCurrency,
+      rpcUrls: chainInformation.urls,
+      blockExplorerUrls: chainInformation.blockExplorerUrls,
+    };
+  } else {
+    return chainId;
+  }
+}
+
+const getInfuraUrlFor = (network: string) =>
+  process.env.infuraKey
+    ? `https://${network}.infura.io/v3/${process.env.infuraKey}`
+    : undefined;
+const getAlchemyUrlFor = (network: string) =>
+  process.env.alchemyKey
+    ? `https://${network}.alchemyapi.io/v2/${process.env.alchemyKey}`
+    : undefined;
+
+type ChainConfig = {
+  [chainId: number]: BasicChainInformation | ExtendedChainInformation;
+};
+
+export const MAINNET_CHAINS: ChainConfig = {
+  1: {
+    urls: [
+      getInfuraUrlFor('mainnet'),
+      getAlchemyUrlFor('eth-mainnet'),
+      'https://cloudflare-eth.com',
+    ].filter(Boolean),
+    name: 'Mainnet',
+  },
+  10: {
+    urls: [
+      getInfuraUrlFor('optimism-mainnet'),
+      'https://mainnet.optimism.io',
+    ].filter(Boolean),
+    name: 'Optimism',
+    nativeCurrency: ETH,
+    blockExplorerUrls: ['https://optimistic.etherscan.io'],
+  },
+  42161: {
+    urls: [
+      getInfuraUrlFor('arbitrum-mainnet'),
+      'https://arb1.arbitrum.io/rpc',
+    ].filter(Boolean),
+    name: 'Arbitrum One',
+    nativeCurrency: ETH,
+    blockExplorerUrls: ['https://arbiscan.io'],
+  },
+  137: {
+    urls: [
+      getInfuraUrlFor('polygon-mainnet'),
+      'https://polygon-rpc.com',
+    ].filter(Boolean),
+    name: 'Polygon Mainnet',
+    nativeCurrency: MATIC,
+    blockExplorerUrls: ['https://polygonscan.com'],
+  },
+  42220: {
+    urls: ['https://forno.celo.org'],
+    name: 'Celo',
+    nativeCurrency: CELO,
+    blockExplorerUrls: ['https://explorer.celo.org'],
+  },
+};
+
+export const TESTNET_CHAINS: ChainConfig = {
+  5: {
+    urls: [getInfuraUrlFor('goerli')].filter(Boolean),
+    nativeCurrency: ETH,
+    name: 'Görli',
+  },
+  420: {
+    urls: [
+      getInfuraUrlFor('optimism-goerli'),
+      'https://goerli.optimism.io',
+    ].filter(Boolean),
+    name: 'Optimism Goerli',
+    nativeCurrency: ETH,
+    blockExplorerUrls: ['https://goerli-explorer.optimism.io'],
+  },
+  421613: {
+    urls: [
+      getInfuraUrlFor('arbitrum-goerli'),
+      'https://goerli-rollup.arbitrum.io/rpc',
+    ].filter(Boolean),
+    name: 'Arbitrum Goerli',
+    nativeCurrency: ETH,
+    blockExplorerUrls: ['https://testnet.arbiscan.io'],
+  },
+  80001: {
+    urls: [getInfuraUrlFor('polygon-mumbai')].filter(Boolean),
```

Comment:
Same for polygon https://github.com/blocto/blocto-sdk/pull/239#discussion_r1311103274